web/pdf_attachment_viewer.js

```diff
@@ -49,152 +46,144 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
-        source: this,
-        attachmentsCount: attachmentsCount,
-      });
+    if (!keepRenderedCapability) {
+      // NOTE: The *only* situation in which the `_renderedCapability` should
+      //       not be replaced is when appending file attachment annotations.
+      this._renderedCapability = createPromiseCapability();
+    }
+  }
 
-      this._renderedCapability.resolve();
-    },
-
-    /**
-     * @private
-     */
-    _bindPdfLink:
-        function PDFAttachmentViewer_bindPdfLink(button, content, filename) {
-      var blobUrl;
-      button.onclick = function() {
-        if (!blobUrl) {
-          blobUrl = createObjectURL(
-            content, 'application/pdf', PDFJS.disableCreateObjectURL);
-        }
-        var viewerUrl;
-        if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
-          // The current URL is the viewer, let's use it and append the file.
-          viewerUrl = '?file=' + encodeURIComponent(blobUrl + '#' + filename);
-        } else if (PDFJSDev.test('CHROME')) {
-          // In the Chrome extension, the URL is rewritten using the history API
-          // in viewer.js, so an absolute URL must be generated.
-          // eslint-disable-next-line no-undef
-          viewerUrl = chrome.runtime.getURL('/content/web/viewer.html') +
-            '?file=' + encodeURIComponent(blobUrl + '#' + filename);
-        } else {
-          // Let Firefox's content handler catch the URL and display the PDF.
-          // In Firefox PDFJS.disableCreateObjectURL is always false, so
-          // blobUrl is always a blob:-URL and never a data:-URL.
-          viewerUrl = blobUrl + '?' + encodeURIComponent(filename);
-        }
-        window.open(viewerUrl);
-        return false;
-      };
-    },
-
-    /**
-     * @private
-     */
-    _bindLink:
-        function PDFAttachmentViewer_bindLink(button, content, filename) {
-      button.onclick = function downloadFile(e) {
-        this.downloadManager.downloadData(content, filename, '');
-        return false;
-      }.bind(this);
-    },
-
-    /**
-     * @param {PDFAttachmentViewerRenderParameters} params
-     */
-    render: function PDFAttachmentViewer_render(params) {
-      params = params || {};
-      var attachments = params.attachments || null;
-      var attachmentsCount = 0;
-
-      if (this.attachments) {
-        var keepRenderedCapability = params.keepRenderedCapability === true;
-        this.reset(keepRenderedCapability);
-      }
-      this.attachments = attachments;
+  /**
+   * @private
+   */
+  _dispatchEvent(attachmentsCount) {
+    this.eventBus.dispatch('attachmentsloaded', {
+      source: this,
+      attachmentsCount,
+    });
 
-      if (!attachments) {
-        this._dispatchEvent(attachmentsCount);
-        return;
+    this._renderedCapability.resolve();
+  }
+
+  /**
+   * @private
+   */
+  _bindPdfLink(button, content, filename) {
+    var blobUrl;
+    button.onclick = () => {
```

Comment:
Nit: Unless I'm missing something, I don't think that you need an arrow function here (no `this` in the function scope).